node/src/components/deploy_acceptor.rs

```diff
@@ -100,121 +183,614 @@ impl DeployAcceptor {
-    fn account_verification<REv: ReactorEventT>(
+    fn handle_get_account_result<REv: ReactorEventT>(
         &mut self,
         effect_builder: EffectBuilder<REv>,
-        deploy: Box<Deploy>,
-        source: Source<NodeId>,
-        account_key: Key,
-        verified: Option<bool>,
-        maybe_responder: Option<Responder<Result<(), Error>>>,
+        event_metadata: EventMetadata,
+        prestate_hash: Digest,
+        maybe_account: Option<Account>,
         verification_start_timestamp: Timestamp,
     ) -> Effects<Event> {
-        let mut effects = Effects::new();
+        match maybe_account {
+            None => {
+                let account_hash = event_metadata.deploy().header().account().to_account_hash();
+                let error = Error::InvalidDeployParameters {
+                    prestate_hash,
+                    failure: DeployParameterFailure::NonexistentAccount { account_hash },
+                };
+                debug!(
+                    ?account_hash,
+                    "nonexistent account associated with the deploy"
+                );
+                self.handle_invalid_deploy_result(
+                    effect_builder,
+                    event_metadata,
+                    error,
+                    verification_start_timestamp,
+                )
+            }
+            Some(account) => {
+                let authorization_keys = event_metadata
+                    .deploy()
+                    .approvals()
+                    .iter()
+                    .map(|approval| approval.signer().to_account_hash())
+                    .collect();
+                if !account.can_authorize(&authorization_keys) {
+                    let error = Error::InvalidDeployParameters {
+                        prestate_hash,
+                        failure: DeployParameterFailure::InvalidAssociatedKeys,
+                    };
+                    debug!(?authorization_keys, "account authorization invalid");
+                    return self.handle_invalid_deploy_result(
+                        effect_builder,
+                        event_metadata,
+                        error,
+                        verification_start_timestamp,
+                    );
+                }
+                if !account.can_deploy_with(&authorization_keys) {
+                    let error = Error::InvalidDeployParameters {
+                        prestate_hash,
+                        failure: DeployParameterFailure::InsufficientDeploySignatureWeight,
+                    };
+                    debug!(?authorization_keys, "insufficient deploy signature weight");
+                    return self.handle_invalid_deploy_result(
+                        effect_builder,
+                        event_metadata,
+                        error,
+                        verification_start_timestamp,
+                    );
+                }
+                return if event_metadata.source().from_client() {
```

Comment:
Can we get rid of these `return if`s ? They're confusing to read.

Reply:
Removed